index.js

```diff
@@ -94,6 +94,7 @@ class ReadStream extends MiniPass {
     if (!this[_reading]) {
       this[_reading] = true
       const buf = this[_makeBuf]()
+      if (buf.length === 0) return this[_onread](null, 0, buf)
```

Comment:
Zalgo?  This calls onread immediately instead of deferring until after the read operation.